File: merge-two-sorted-lists/scala/ListNode.scala
class ListNode(_x: Int = 0, _next: ListNode = null) {
    var next: ListNode = _next
    var x: Int = _x
}

object ListNode {
    def toList(list: ListNode): List[Int] = {
        if (list == null) {
            List()
        }
        else if (list.next != null) {
            list.x :: toList(list.next)
        }
        else {
            List(list.x)
        }
    }
}

File: merge-two-sorted-lists/cs/ListNode.cs
public class ListNode
{
    public int val;
    public ListNode next;
    public ListNode(int val = 0, ListNode next = null)
    {
        this.val = val;
        this.next = next;
    }
}

public static class ListNodeUtils
{
    public static List<int> ToList(this ListNode node)
    {
        if (node == null)
        {
            return [];
        }
        else if (node.next != null)
        {
            List<int> a = [node.val];
            List<int> b = ToList(node.next);
            return [.. a, .. b];
        }
        else
        {
            return [node.val];
        }
    }
}